flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -310,6 +313,10 @@ public void open() throws CatalogException {
                             "Configured default database %s doesn't exist in catalog %s.",
                             getDefaultDatabase(), getName()));
         }
+
+        if (cacheTable == null) {
```

Comment:
Would you please help me understand why lazy init is required in this case? Afaik, without calling the open() the catalog can not be used.

Reply:
Hi , this has no special design and we can directly use cache without going through open method.